### ISSUE TITLE
boost/optional.hpp file not found - when building on os x

### ISSUE THREAD

**Issue by rojomisin** [BeamMW/beam#513]
**Bug title**
Express the problem in a single concise sentence

**Bug description**
`cmake -DCMAKE_BUILD_TYPE=Release .` succeeds
`make -j4` fails

**To Reproduce**
Steps to reproduce the behaviour:
1. On OS X Mojave 10.14.3 following the build instructions, cmake fails
`brew install openssl boost cmake qt5`
`export OPENSSL_ROOT_DIR="/usr/local/opt/openssl"`
`export PATH=/usr/local/opt/qt/bin:$PATH`
```
**Current behaviour**
Scanning dependencies of target quazip_static
[  8%] Building CXX object 3rdparty/quazip/CMakeFiles/quazip_static.dir/quazip/JlCompress.cpp.o
[  8%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/uv-data-getter-setters.c.o
[  9%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/version.c.o
[  9%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/async.c.o
[  9%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/core.c.o
[  9%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/dl.c.o
[ 10%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/fs.c.o
[ 11%] Building CXX object pow/CMakeFiles/pow.dir/__/3rdparty/crypto/equihash_impl.cpp.o
[ 11%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/getaddrinfo.c.o
[ 11%] Building CXX object 3rdparty/quazip/CMakeFiles/quazip_static.dir/quazip/qioapi.cpp.o
/Users/rmicone/Dev/beam/3rdparty/crypto/equihash_impl.cpp:27:10: fatal error: 'boost/optional.hpp' file not found
#include <boost/optional.hpp>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
[ 11%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/getnameinfo.c.o
make[2]: *** [pow/CMakeFiles/pow.dir/__/3rdparty/crypto/equihash_impl.cpp.o] Error 1
make[1]: *** [pow/CMakeFiles/pow.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 11%] Building CXX object 3rdparty/quazip/CMakeFiles/quazip_static.dir/quazip/quaadler32.cpp.o
[ 12%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/loop-watcher.c.o
[ 12%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/loop.c.o
[ 12%] Building CXX object 3rdparty/quazip/CMakeFiles/quazip_static.dir/quazip/quacrc32.cpp.o
[ 12%] Building C object 3rdparty/libuv/CMakeFiles/uv_a.dir/src/unix/pipe.c.o
```
**Expected behaviour**

**Platform and build**
* Windows, Mac, Linux, Android or iOS
* OS version
* BEAM Build number

**Configuration, log, memory dump files (if applicable)**
above

**Log file(s) (if applicable)**
above

**Screenshots (if applicable)**

**Comment by Sergei-Beam** [comment]
@nesbox Can you help there?

**Comment by nesbox** [comment]
yes, I reproduced it on 10.14 too

**Comment by nesbox** [comment]
fixed here b71fd88 on `master`
@rojomisin pls pull and generate project with cmake again

**Comment by rojomisin** [comment]
fixed thx!
👍 🙏
